packages/bigquery/src/table.js

```diff
@@ -188,42 +188,64 @@ Table.createSchemaFromString_ = function(str) {
  * @return {array} Fields using their matching names from the table's schema.
  */
 Table.mergeSchemaWithRows_ = function(schema, rows) {
-  return rows.map(mergeSchema).map(flattenRows);
+  return arrify(rows).map(mergeSchema).map(flattenRows);
 
   function mergeSchema(row) {
     return row.f.map(function(field, index) {
       var schemaField = schema.fields[index];
       var value = field.v;
 
-      switch (schemaField.type) {
-        case 'BOOLEAN': {
-          value = value === 'true';
-          break;
-        }
-        case 'FLOAT': {
-          if (!is.nil(value)) {
-            value = parseFloat(value);
-          }
-          break;
-        }
-        case 'INTEGER': {
-          if (!is.nil(value)) {
-            value = parseInt(value, 10);
-          }
-          break;
-        }
-        case 'TIMESTAMP': {
-          value = new Date(value * 1000);
-          break;
-        }
+      var fieldObject = {};
+
+      if (value === null) {
+        fieldObject[schemaField.name] = null;
+        return fieldObject;
+      }
+
+      if (schemaField.mode === 'REPEATED') {
+        value = value.map(function(val) {
+          return convert(schemaField, val.v);
+        });
+      } else {
+        value = convert(schemaField, value);
       }
 
-      var fieldObject = {};
       fieldObject[schemaField.name] = value;
       return fieldObject;
     });
   }
 
+  function convert(schemaField, value) {
+    switch (schemaField.type) {
+      case 'BOOLEAN': {
+        value = value === 'true';
+        break;
+      }
+      case 'FLOAT': {
+        if (!is.nil(value)) {
```

Comment:
you may instead do once nil check at beginning of convert function; while the docs saying `query result ARRAYs can never have NULL elements`

https://cloud.google.com/bigquery/sql-reference/data-types#array-type

in practice, if I try to have null in array, it ends up an error:

```js
node ./bigquery/queries.js sync 'SELECT ARRAY[1,3,2,NULL]'
{ err: 
   { ApiError: Array 'f0_' cannot have a null element

     errors: 
      [ { domain: 'global',
          reason: 'invalidQuery',
          message: 'Array \'f0_\' cannot have a null element',
          locationType: 'other',
          location: 'query' } ],
     response: undefined,
     message: 'Array \'f0_\' cannot have a null element' },
  rows: null,
  nextQuery: null,
  apiResponse: 
   { error: 
      { errors: 
         [ { domain: 'global',
             reason: 'invalidQuery',
             message: 'Array \'f0_\' cannot have a null element',
             locationType: 'other',
             location: 'query' } ],
        code: 400,
        message: 'Array \'f0_\' cannot have a null element' } } }

```